clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1333,7 +1332,14 @@ boolean updateAssignmentMetadataIfNeeded(final Timer timer) {
      *
      * @throws org.apache.kafka.clients.consumer.CommitFailedException if the commit failed and cannot be retried.
      *             This can only occur if you are using automatic group management with {@link #subscribe(Collection)},
-     *             or if there is an active group with the same groupId which is using group management.
+     *             or if there is an active group with the same groupId which is using group management. In such cases,
+     *             when you are trying to commit to partitions that is no longer assigned to this consumer because the
+     *             consumer is for example no longer part of the group this exception would be thrown.
+     * @throws org.apache.kafka.clients.consumer.RetriableCommitFailedException if the commit failed but can be retriable.
```

Comment:
Maybe "can be _retried_"?

Reply:
"Can be retriable" does seem to imply that in some circumstances it can be retried, while in others it _can't_ be retried at all. "Can be retried" implies it's up to the user whether they want to retry